dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/imports/programrule/ProgramRuleAssignActionTest.java

```diff
@@ -119,9 +130,62 @@ void shouldNotImportWithWarningWhenAttributeWithSameValueIsAssignedByAssignRule(
     assertHasOnlyWarnings(importReport, E1310);
   }
 
+  @Test
+  void shouldImportEventWithLastDateAndCorrectlyAssignPreviousEventDataValue() throws IOException {
+    TrackerImportParams params = new TrackerImportParams();
+    TrackerObjects trackerObjects =
+        fromJson("tracker/programrule/three_events_with_different_dates.json");
+    params.setImportStrategy(TrackerImportStrategy.CREATE_AND_UPDATE);
+
+    trackerImportService.importTracker(params, trackerObjects);
+
+    assignPreviousEventProgramRule();
+
+    trackerObjects = fromJson("tracker/programrule/event_with_last_date.json");
+
+    ImportReport importReport = trackerImportService.importTracker(params, trackerObjects);
+    assertHasOnlyWarnings(importReport, E1308);
+
+    Event event = manager.get(Event.class, "D9PbzJY8bZZ");
+
+    List<String> eventDataValues =
+        event.getEventDataValues().stream()
+            .filter(dv -> dv.getDataElement().equals("DATAEL00002"))
+            .map(dv -> dv.getValue())
```

Comment:
a method reference should work right?

Reply:
Yes, that should have been caught by Sonar :thinking: